### PR TITLE
WARN when using app.config to configure the Gateway

### PR DESCRIPTION
**Pull request by andreasohlund**
Will do upgrade guide and doco for `.TransactionTimeout` once merged

### CODE REVIEW

File: src/NServiceBus.Gateway/Settings/GatewaySettings.cs [hunk 1]
```diff
@@ -143,6 +143,17 @@ public void AddReceiveChannel(string address, string type = "http", int maxConcu
             });
         }
 
+        /// <summary>
+        /// Configures the transaction timeout to use when transmitting messages to remote sites.
+        /// </summary>
+        /// <param name="timeout"></param>
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/Settings/GatewaySettings.cs]: 🔥

File: src/NServiceBus.Gateway/Settings/GatewaySettings.cs [hunk 2]
```diff
@@ -155,7 +166,10 @@ public void AddReceiveChannel(string address, string type = "http", int maxConcu
 #if NET452
             var configSection = GetConfigSection(settings);
 
-
+            if (configSection?.TransactionTimeout != null)
+            {
+                logger.WarnFormat("The ability to specify transaction timeout via the GatewayConfig config section will be removed in v4. Instead use the `EndpointConfiguration.Gateway().TransactionTimeout(...)` API.");
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/Settings/GatewaySettings.cs]: not sure tbh, but imho `Use the `EndpointConfiguration.Gateway().TransactionTimeout(...)` API instead.` reads better?

File: src/NServiceBus.Gateway/Settings/GatewaySettings.cs [hunk 3]
```diff
@@ -177,6 +191,8 @@ internal static List<Site> GetConfiguredSites(ReadOnlySettings settings)
                 return new List<Site>();
             }
 
+            logger.WarnFormat("The ability to specify sites via the GatewayConfig config section will be removed in v4. Instead use the `EndpointConfiguration.Gateway().AddSite(...)` API.");
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/Settings/GatewaySettings.cs]: same as above

File: src/NServiceBus.Gateway/Settings/GatewaySettings.cs [hunk 4]
```diff
@@ -208,6 +224,8 @@ internal static List<ReceiveChannel> GetConfiguredChannels(ReadOnlySettings sett
                 return new List<ReceiveChannel>();
             }
 
+            logger.WarnFormat("The ability to specify receive channels via the GatewayConfig config section will be removed in v4. Instead use the `EndpointConfiguration.Gateway().AddReceiveChannel(...)` API.");
```
Comment (timbussmann) [inline src/NServiceBus.Gateway/Settings/GatewaySettings.cs]: same as above

**Comment by andreasohlund** [comment]
@timbussmann addressed your feedback

**Comment by timbussmann** [comment]
there seems to be something off with the build I think? @Particular/deployment-maintainers @andreasohlund

**Comment by dbelcham** [comment]
@timbussmann PR #79 fixes the issues with the test runs here. Once that gets merged this PR should go green too

**Comment by andreasohlund** [comment]
Is now running green

**Comment by andreasohlund** [comment]
Doco PR started https://github.com/Particular/docs.particular.net/pull/3304

**Pull request merged**